lib/index.ts

```diff
@@ -148,39 +148,32 @@ export const imagePath = (app: string, file: string) => {
  * @return {string} URL with webroot for a file in an app
  */
 export const generateFilePath = (app: string, type: string, file: string) => {
-	const isCore = window?.OC?.coreApps?.indexOf(app) !== -1
+	const isCore = window?.OC?.coreApps?.includes(app)
+	const isPHP = file.slice(-3) === 'php'
 	let link = getRootUrl()
-	if (file.substring(file.length - 3) === 'php' && !isCore) {
-		link += '/index.php/apps/' + app
+	if (isPHP && !isCore) {
+		link += `/index.php/apps/${app}`
 		if (file !== 'index.php') {
 			link += '/'
 			if (type) {
 				link += encodeURI(type + '/')
 			}
 			link += file
 		}
-	} else if (file.substring(file.length - 3) !== 'php' && !isCore) {
+	} else if (!isPHP && !isCore) {
 		link = getAppRootUrl(app)
 		if (type) {
-			link += '/' + type + '/'
+			link += `/${type}/`
 		}
-		if (link.substring(link.length - 1) !== '/') {
+		if (link.at(-1) !== '/') {
```

Comment:
`.at` supports negative indices (like in most other languages) which are the index from the end, meaning -1 is the last char.
Reduces the complexity of this statement.